### PR TITLE
Add Managers creation

### PR DESCRIPTION
**Pull request by VStetsyshyn**

### CODE REVIEW

File: app/views/devise/registrations/edit.html.erb [hunk 1]
```diff
@@ -20,18 +21,25 @@
...
+
+  <% if current_user.admin?%>
+    <div class="form-group"> 
+      <%= f.button :submit => "", name: "id", value: @user.id, class: 'btn btn-danger'%>
+    </div>
+  <% else %>
+    <div class="form-group">
+      <%= f.submit t('.update'), class: 'btn btn-primary' %>
+    </div>
+  <% end %>
```
Comment (BohdanChaban) [inline app/views/devise/registrations/edit.html.erb]: Blocks on lines 24-41 can be combined in one if-else operator

File: spec/factories/user_factory.rb [hunk 2]
```diff
@@ -6,6 +6,12 @@
     role { 'customer' }
     password { '12345678' }
 
+    #   trait :man do
+    # email { 'manager@gmail.com' }
+    # role { 'manager' }
+    # password { '12345678' }
+    # end
+
```
Comment (BohdanChaban) [inline spec/factories/user_factory.rb]: ???

File: app/assets/stylesheets/registrations.scss [hunk 3]
```diff
@@ -0,0 +1,3 @@
+.form-group {
+	width:20%;
+}
```
Comment (BohdanChaban) [inline app/assets/stylesheets/registrations.scss]: missed end line

File: app/views/accounts/show.html.erb [hunk 4]
```diff
@@ -14,6 +14,6 @@
     </p>
     </div>
 </div>
-<%= link_to '<button type="button" class="btn btn-warning">Edit</button>'.html_safe, edit_account_path(@account) %>
+<%= link_to '<button type="button" class="btn btn-danger">Edit</button>'.html_safe, edit_account_path(@account) %>
```
Comment (BohdanChaban) [inline app/views/accounts/show.html.erb]: Pls use `btn-warning` class

File: app/views/accounts/index.html.erb [hunk 5]
```diff
@@ -21,12 +21,18 @@
...
       <% end %>
     </tbody>
   </table>
 <% else %>
    <div>There are no accounts</div>
 <% end %>
-<%= link_to '<button type="button" class="btn btn-warning">New Account</button>'.html_safe, new_account_path %>
 
+<% if params[:role] == "manager" %>
+<%= link_to '<button type="button" class="btn btn-success">New Manager</button>'.html_safe, new_user_registration_path %>
```
Comment (BohdanChaban) [inline app/views/accounts/index.html.erb]: Pls use `btn-warning` class

File: app/views/accounts/index.html.erb [hunk 6]
```diff
@@ -21,12 +21,18 @@
           <td><%= account.surname %></td>
           <td><%= account.address %></td>
           <td><%= account.mobile_number %></td>
+          <td><%= link_to '<button type="button" class="btn btn-light">Show</button>'.html_safe, account %></td>
```
Comment (BohdanChaban) [inline app/views/accounts/index.html.erb]: Pls use `btn-warning` class

File: app/views/accounts/index.html.erb [hunk 7]
```diff
@@ -21,12 +21,18 @@
...
+          <% if params[:role] == "manager" && current_user.admin? %>
+            <td><%= link_to '<button type="button" class="btn btn-dark">Edit</button>'.html_safe,edit_user_registration_path(:id => account.user_id) %></td>
```
Comment (BohdanChaban) [inline app/views/accounts/index.html.erb]: Pls use `btn-warning` class

File: app/views/devise/registrations/edit.html.erb [hunk 8]
```diff
@@ -2,11 +2,12 @@
 
 <%= form_for(resource, as: resource_name, url: registration_path(resource_name), html: { method: :put }) do |f| %>
   <%= bootstrap_devise_error_messages! %>
-
-  <div class="form-group">
-    <%= f.label :email %>
-    <%= f.email_field :email, autofocus: true, autocomplete: 'email', class: 'form-control' %>
-  </div>
+  <% if !current_user.admin?%>
```
Comment (BohdanChaban) [inline app/views/devise/registrations/edit.html.erb]: `unless`

File: config/routes.rb [hunk 9]
```diff
@@ -7,6 +7,11 @@
   resources :articles
 
   devise_for :users, controllers: { registrations: 'registrations' }
+
+  namespace :admin do
+    get '', to: 'dashboard#index', as: '/'
+    resources :users
```
Comment (BohdanChaban) [inline config/routes.rb]: u need all resources?

File: app/views/devise/registrations/edit.html.erb [hunk 10]
```diff
@@ -20,18 +21,25 @@
...
-
-  <div class="form-group">
-    <%= f.submit t('.update'), class: 'btn btn-warning' %>
-  </div>
 <% end %>
 
-<p><%= t('.unhappy') %>? <%= link_to t('.cancel_my_account'), registration_path(resource_name), data: { confirm: t('.are_you_sure') }, method: :delete %>.</p>
-
-<%= link_to t('.back'), :back %>
+<%= link_to t('.back'), :back, class: "btn btn-light" %>
```
Comment (BohdanChaban) [inline app/views/devise/registrations/edit.html.erb]: Pls use `btn-warning` class

File: app/views/devise/registrations/edit.html.erb [hunk 11]
```diff
@@ -20,18 +21,25 @@
...
+    </div>
+  <% end %>
+
+  <% if current_user.admin?%>
+    <div class="form-group"> 
+      <%= f.button :submit => "", name: "id", value: @user.id, class: 'btn btn-danger'%>
+    </div>
+  <% else %>
+    <div class="form-group">
+      <%= f.submit t('.update'), class: 'btn btn-primary' %>
```
Comment (BohdanChaban) [inline app/views/devise/registrations/edit.html.erb]: Pls use `btn-warning` class

File: app/views/devise/registrations/edit.html.erb [hunk 12]
```diff
@@ -20,18 +21,25 @@
...
+      <%= f.label :current_password %>
+      <%= f.password_field :current_password, autocomplete: 'current-password', class: 'form-control' %>
+
+      <small class="form-text text-muted"><%= t('.we_need_your_current_password_to_confirm_your_changes') %></small>
+    </div>
+  <% end %>
+
+  <% if current_user.admin?%>
+    <div class="form-group"> 
+      <%= f.button :submit => "", name: "id", value: @user.id, class: 'btn btn-danger'%>
```
Comment (BohdanChaban) [inline app/views/devise/registrations/edit.html.erb]: Pls use `btn-warning` class

File: config/routes.rb (hunk 9 above)
Comment (VStetsyshyn) [reply config/routes.rb]: not need this at all

File: app/views/devise/registrations/edit.html.erb [hunk 13]
```diff
@@ -20,18 +21,22 @@
...
+  <% if current_user.admin?%>
+    <div class="form-group"> 
+      <%= f.button :submit => "", name: "id", value: @user.id, class: 'btn btn-warning'%>
+    </div>
+  <% else %>
+    <div class="form-group">
+      <%= f.label :current_password %>
+      <%= f.password_field :current_password, autocomplete: 'current-password', class: 'form-control' %>
 
     <small class="form-text text-muted"><%= t('.we_need_your_current_password_to_confirm_your_changes') %></small>
```
Comment (BohdanChaban) [inline app/views/devise/registrations/edit.html.erb]: missed indent

File: app/views/accounts/index.html.erb [hunk 14]
```diff
@@ -21,12 +21,17 @@
...
       <% end %>
     </tbody>
   </table>
 <% else %>
    <div>There are no accounts</div>
 <% end %>
-<%= link_to '<button type="button" class="btn btn-warning">New Account</button>'.html_safe, new_account_path %>
 
+<% if params[:role] == "manager" %>
+<%= link_to '<button type="button" class="btn btn-warning">New Manager</button>'.html_safe, new_user_registration_path %>
```
Comment (BohdanChaban) [inline app/views/accounts/index.html.erb]: missed indent

File: app/controllers/registrations_controller.rb [hunk 15]
```diff
@@ -1,6 +1,54 @@
...
+      end
+    else
+      super
+    end
+  end
+
   protected
 
+  before_action :authorize_user!, only: %i[new create]
+  skip_before_action :require_no_authentication, only: %i[new create]
```
Comment (BohdanChaban) [inline app/controllers/registrations_controller.rb]: callbacks should be at the top of class

**Pull request merged**